.github/workflows/tox-tests.yml

```diff
@@ -0,0 +1,25 @@
+name: Tox Tests
+
+on:
+  pull_request:
+
+jobs:
+  tox-tests:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Set up Python 3.13
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.13'
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install tox
+
+    - name: Run tox environments
```

Comment:
you can use uv action here as well and use uv-tox for better performance
 uv tool install tox --with tox-uv